tests/issues.test.js

```diff
@@ -0,0 +1,170 @@
+            .send({ description: 'Lorem ipsum', status: true })
+            .end((err, res) => {
+                res.should.have.status(201);
+                res.should.be.json;
+                res.body.should.be.a('object');
+                res.body.should.have
+                    .property('message')
+                    .equal('Issue created successfully');
+                res.body.should.have.property('createdIssue');
+                res.body.createdIssue.should.have.keys(
+                    'id',
+                    'description',
+                    'status',
+                    'comments',
+                    'resourceUrl',
+                    'issueFiles'
+                );
+                res.body.createdIssue.description.should.equal('Lorem ipsum');
+                res.body.createdIssue.status.should.equal(true);
+                done();
+            });
+    });
+
+    it('should not post issue without required field', done => {
+        chai.request(app)
+            .post('/issues')
+            .send()
+            .end((err, res) => {
+                res.should.have.status(500);
+                res.body.should.have.property('error');
+                done();
+            });
+    });
+
+    it('should update issue by the given id', done => {
+        var issue = new Issue({
+            _id: new mongoose.Types.ObjectId(),
+            description: 'Something ',
+            status: false
+        });
+        issue.save((err, issue) => {
+            chai.request(app)
+                .patch('/issues/' + issue._id)
+                .send([{ propName: 'description', value: 'Something new' }])
+                .end((err, res) => {
+                    res.should.have.status(200);
+                    res.should.be.json;
+                    res.body.should.be.a('object');
+                    res.body.should.have.property('message');
+                    res.body.message.should.equal('Issue is updated');
+                    done();
+                });
+        });
+    });
+
+    it('should delete issue by the given id', done => {
+        var issue = new Issue({
+            _id: new mongoose.Types.ObjectId(),
+            description: 'Something',
+            status: false
+        });
+        issue.save((err, issue) => {
+            chai.request(app)
+                .delete('/issues/' + issue._id)
+                .end((err, res) => {
+                    res.should.have.status(200);
+                    res.should.be.json;
+                    res.body.should.be.a('object');
+                    res.body.should.have.property('message');
+                    res.body.message.should.equal('Issue is deleted!');
+                    done();
+                });
+        });
+    });
+});
+
+after(() => {
+    Issue.collection.drop();
+    return mongoose.disconnect(() => {
+        mongoose.models = {};
+        mongoose.modelSchemas = {};
+        mongoose.connection.close();
+    });
+});
```

Comment:
Great tests! This is great progress... Some feedback:
- Look into `before`, `beforeEach` and `after` functionalities.
- Right now you are dropping collections only after all the tests are done. Each test case (`it`) should be ran independently, its execution should not be dependent on the previous test.
- `describe` - sets the context "This is what I'm testing". You shouldn't limit on using it only at the beginning of testing. You can also use it to set the context like "request body has invalid parameters"
- `before` - you are preparing your test check, for example setting up data. In your case those are these `issue.save`s that you have
- `it` - this is where you actually do the check